logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -409,14 +409,14 @@ private void updateOldestSegmentReference() throws IOException {
      * */
     private static Optional<Timestamp> readTimestampOfLastEventInSegment(Path segmentPath) throws IOException {
         final int lastBlockId = (int) Math.ceil(((Files.size(segmentPath) - VERSION_SIZE) / (double) BLOCK_SIZE)) - 1;
-        byte[] eventBytes;
+        byte[] eventBytes = null;
         try (RecordIOReader recordReader = new RecordIOReader(segmentPath)) {
             int blockId = lastBlockId;
-            do {
+            while (eventBytes == null && blockId >= 0) { // no event present in last block, try with the one before
```

Comment:
This method is used only by `updateOldestSegmentReference`

https://github.com/elastic/logstash/blob/58abffce330533c8c6852e6a7f7fe3ae5d9d34fa/logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java#L390-L398

So should be better to filter the segments with length equals to 1 like we did in #14605 ?

From
```java
oldestSegmentPath = listSegmentPaths(this.queuePath).sorted().findFirst();
```
to 
```java
oldestSegmentPath = listSegmentPaths(this.queuePath)
   .filter(p -> p.toFile().length() > 1) // take the files that have content to process
   .sorted()
   .findFirst();
```

Reply:
yes, it is better 👍